src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -456,52 +456,26 @@ private CommandInfo GetNextFromPath()
                     "Trying to resolve the path as an PSPath");
 
                 // Find the match if it is.
-
-                Collection<string> resolvedPaths = new Collection<string>();
-
-                try
-                {
-                    Provider.CmdletProvider providerInstance;
-                    ProviderInfo provider;
-                    resolvedPaths =
-                        _context.LocationGlobber.GetGlobbedProviderPathsFromMonadPath(_commandName, false, out provider, out providerInstance);
-                }
-                catch (ItemNotFoundException)
+                // Try literal path resolution if it is set to run first
+                if (_commandResolutionOptions.HasFlag(SearchResolutionOptions.ResolveLiteralThenPathPatterns))
                 {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "The path could not be found: {0}",
-                        _commandName);
-                }
-                catch (DriveNotFoundException)
-                {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "A drive could not be found for the path: {0}",
-                        _commandName);
+                    string path = GetNextLiteralPathThatExists(_commandName, out _);
+                    return GetInfoFromPath(path);
```

Comment:
GetInfoFromPath can return null, but we return here and not continue looking with patterns.  But the flag set suggests that patterns are searched next.  Do we expect to return null here?  Can you add a comment if this is expected behavior?

Reply:
```suggestion
                    // This can return null, but that is expected.
                    // The searcher will continue, if it can.
                    return GetInfoFromPath(path);
```

Reply:
I think I misunderstood your comment.  We should not call GetInfoFromPath if path is null.  I fixed the code.